weblab/datasets/tests/test_views.py

```diff
@@ -120,6 +120,87 @@ def test_add_delete_add_file(self, logged_in_user, client, helpers, public_proto
                 assert fp.read() == file_map[f.name]
 
 
+@pytest.mark.django_db
+class TestDatasetDeletion:
+
+    def test_owner_can_delete_dataset(
+            self, helpers, logged_in_user, public_protocol , client
+    ):
+        helpers.add_permission(logged_in_user, 'create_dataset', Dataset)
+        dataset = recipes.dataset.make(author=logged_in_user, name='mydataset', protocol=public_protocol)
+        file_name = 'mydataset.csv'
+        file_contents = b'my test dataset'
+        recipes.dataset_file.make(  # Uploaded then deleted
+            dataset=dataset,
+            upload=SimpleUploadedFile(file_name, file_contents),
+            original_name=file_name,
+        )
+        client.post(
+            '/datasets/%d/addfiles' % dataset.pk,
+            data={
+                'filename[]': ['uploads/' + file_name],
+                'delete_filename[]': [],
+                'mainEntry': [file_name],
+            },
+        )
+        response = client.post('/datasets/%d/delete' % dataset.pk)
```

Comment:
Perhaps add `assert dataset.archive_path.exists()` before this line to check it really accomplishes something?

Reply:
Yes - thanks